### ISSUE TITLE
Lambda and Subclass layer

### ISSUE THREAD

**Issue by akshitadixit** [tensorflow/tensorflow#49901]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- Using Google colab
- TensorFlow version: `tensorflow._api.v2.version`
- GPU model and memory: Python Google Compute Engine Backend (2 cores)

This is the warning I get:
<pre>Downloading data from https://storage.googleapis.com/tensorflow/keras-applications/resnet/resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5
94773248/94765736 [==============================] - 1s 0us/step
WARNING:tensorflow:
The following Variables were used a Lambda layer's call (tf.tensordot), but
are not present in its tracked objects:
  <tf.Variable 'dense_1/kernel:0' shape=(2048, 2048) dtype=float32>
It is possible that this is intended behavior, but it is more likely
an omission. This is a strong indication that this layer should be
formulated as a subclassed Layer rather than a Lambda layer.
WARNING:tensorflow:
The following Variables were used a Lambda layer's call (tf.nn.bias_add), but
are not present in its tracked objects:
  <tf.Variable 'dense_1/bias:0' shape=(2048,) dtype=float32>
It is possible that this is intended behavior, but it is more likely
an omission. This is a strong indication that this layer should be
formulated as a subclassed Layer rather than a Lambda layer.
WARNING:tensorflow:
The following Variables were used a Lambda layer's call (tf.tensordot_1), but
are not present in its tracked objects:
  <tf.Variable 'dense/kernel:0' shape=(2048, 5) dtype=float32>
It is possible that this is intended behavior, but it is more likely
an omission. This is a strong indication that this layer should be
formulated as a subclassed Layer rather than a Lambda layer.
WARNING:tensorflow:
The following Variables were used a Lambda layer's call (tf.nn.bias_add_1), but
are not present in its tracked objects:
  <tf.Variable 'dense/bias:0' shape=(5,) dtype=float32>
It is possible that this is intended behavior, but it is more likely
an omission. This is a strong indication that this layer should be
formulated as a subclassed Layer rather than a Lambda layer.</pre>

**Standalone code to reproduce the issue**
<pre>input_tensor = tf.keras.layers.Input(shape=(IMG_SIZE, IMG_SIZE, CANAL))

base = ResNet50(weights="imagenet", include_top=False,input_tensor=input_tensor)
#base.load_weights('resnet50_weights_tf_dim_ordering_tf_kernels_notop.h5')

top = Dense(N_CLASSES, activation='softmax')(
            Dropout(0.5)(
                Dense(2048, activation='relu')(
                    Dropout(0.5)(
                        MaxPooling2D()(base.output)
                    )
                )
            )
        )

model = Model(input_tensor, top)</pre>

Kindly help since this isn't allowing me to train all the pre-existing layers in the model.

N_CLASSES= 5
CANAL = 3
IMG_SIZE = 512

**Comment by tusharvickey1999** [comment]
@akshitadixit kindly mention the values for IMG_SIZE, CANAL, N_CLASSES for which you are getting error. This code is not creating any error in Google colab environment.

**Comment by akshitadixit** [comment]
@tusharvickey1999 updated, added at the end

**Comment by tusharvickey1999** [comment]
> @tusharvickey1999 updated, added at the end

Can you add a link to the Google colab notebook which produces this error since this code is not producing any error.

**Comment by Saduf2019** [comment]
@akshitadixit 
Please refer to the gist shared [here](
https://colab.research.google.com/gist/tilakrayal/6b16407f9485e40a8b80f83efc51b633/49901.ipynb), we do not see any issues.

**Comment by akshitadixit** [comment]
> @akshitadixit
> Please refer to the gist shared [here](https://colab.research.google.com/gist/tilakrayal/6b16407f9485e40a8b80f83efc51b633/49901.ipynb), we do not see any issues.

I'm so sorry for the time you put here. I'll run my thing once again. Though I obviously did not write the error warnings myself, I will recheck. Thank you for the patience.

**Comment by akshitadixit** [comment]
Hey @Saduf2019 I checked out the gist and it seemed to work perfectly, however strangely doesn't yet work on my notebook. Attaching screenshots for reference. Both of them aren't any different but I just tried to be very sure.

![Screenshot from 2021-06-03 16-35-33](https://user-images.githubusercontent.com/56997545/120638095-562aa400-c48d-11eb-969e-640bbf6d6d4a.png)
![Screenshot from 2021-06-03 17-02-30](https://user-images.githubusercontent.com/56997545/120638333-a0138a00-c48d-11eb-846f-67c207517ada.png)

**Comment by Saduf2019** [comment]
@akshitadixit 
Does the warning effect your performance/result compared to what we see on colab, if not we can just ignore the warning.

**Comment by akshitadixit** [comment]
The warning somehow returns a model with zero layers on my notebook, but gives the ideal output on the gist you shared. This is really strange.

**Comment by tusharvickey1999** [comment]
@akshitadixit can you confirm if the warning is result of using the Resnet50 model in your network?

**Comment by Saduf2019** [comment]
@akshitadixit 
You may want to check the configurational differences from colab to your system, and update your system accordingly. Can you try on venv and let us know if the issue exists.

**Comment by akshitadixit** [comment]
Sure I will.

**Comment by akshitadixit** [comment]
@Saduf2019 it works super normal on my local virtual environment, but still doesn't run on my colab notebook :(

[Link](https://colab.research.google.com/drive/19S-0sgYe5xoEU1EEaf1BLQV1io45Y6ey?usp=sharing) to my colab notebook.

**Comment by akshitadixit** [comment]
> @akshitadixit can you confirm if the warning is result of using the Resnet50 model in your network?

Yes it did spring up when I used resnet, but I'm not exactly sure if it is because of that. Never happened before. Although I'm afraid if I have placed my layers somehow incorrectly, but that's the way I have seen it getting done.

**Comment by tusharvickey1999** [comment]
@akshitadixit finally figured out that importing Dense layer from keras was creating the error, importing it from tensorflow.keras makes no error. Its probably due to the fact that resnet50 is pretrained on dense layer of tensorflow and importing keras dense layer creates conflict bw the two.

hope this helps.

**Comment by akshitadixit** [comment]
How does it explain that the exact same code runs well on my local machine?

**Comment by Saduf2019** [comment]
@akshitadixit 
As it works you may move this to close status and open the topic on tensorflow discussion forum or stackoverflow as there is a larger community there who could answer that.

**Comment by akshitadixit** [comment]
Sure thing. Thanks for the time.
